From 071045a3f38c107d244d183d3c60ac2a8183b25c Mon Sep 17 00:00:00 2001
From: Romain Rigaux <romain@cloudera.com>
Date: Fri, 12 Dec 2014 10:22:33 -0800
Subject: [PATCH 0149/1173] HUE-2484 [beeswax] Configure support for Hive
 Server2 LDAP authentication

This makes sure we use plain SASL auth in Hive when we have

<property>
  <name>hive.server2.authentication</name>
  <value>LDAP</value>
</property
---
 .../beeswax/src/beeswax/server/hive_server2_lib.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/apps/beeswax/src/beeswax/server/hive_server2_lib.py b/apps/beeswax/src/beeswax/server/hive_server2_lib.py
index 093f741..6d5fd96 100644
--- a/apps/beeswax/src/beeswax/server/hive_server2_lib.py
+++ b/apps/beeswax/src/beeswax/server/hive_server2_lib.py
@@ -299,7 +299,7 @@ class HiveServerTColumnDesc:
 
 
 class HiveServerClient:
-  HS2_MECHANISMS = {'KERBEROS': 'GSSAPI', 'NONE': 'PLAIN', 'NOSASL': 'NOSASL'}
+  HS2_MECHANISMS = {'KERBEROS': 'GSSAPI', 'NONE': 'PLAIN', 'NOSASL': 'NOSASL', 'LDAP': 'PLAIN'}
 
   def __init__(self, query_server, user):
     self.query_server = query_server
@@ -376,7 +376,7 @@ class HiveServerClient:
       hive_mechanism = hive_site.get_hiveserver2_authentication()
       if hive_mechanism not in HiveServerClient.HS2_MECHANISMS:
         raise Exception(_('%s server authentication not supported. Valid are %s.' % (hive_mechanism, HiveServerClient.HS2_MECHANISMS.keys())))
-      use_sasl = hive_mechanism in ('KERBEROS', 'NONE')
+      use_sasl = hive_mechanism in ('KERBEROS', 'NONE', 'LDAP')
       mechanism = HiveServerClient.HS2_MECHANISMS[hive_mechanism]
       impersonation_enabled = hive_site.hiveserver2_impersonation_enabled()
 
-- 
1.7.9.5

